src/sentry/utils/redis.py

```diff
@@ -142,28 +142,20 @@ def check_cluster_versions(cluster, required, recommended=None, label=None):
     )
 
 
-def load_script(path):
-    cache = WeakKeyDictionary()
-    source = resource_string('sentry', posixpath.join('scripts', path))
-
-    # This changes the argument order of the ``Script.__call__`` method to
-    # encourage using the script with a specific Redis client, rather
-    # than implicitly using the first client that the script was registered
-    # with. (This can prevent lots of bizzare behavior when dealing with
-    # clusters of Redis servers.)
-    def call_script(client, keys, args):
-        """
-        Executes {!r} as a Lua script on a Redis server.
-
-        Takes the client to execute the script on as the first argument,
-        followed by the values that will be provided as ``KEYS`` and ``ARGV``
-        to the script as two sequence arguments.
-        """.format(path)
-        script = cache.get(client)
-        if script is None:
-            script = cache[client] = Script(client, source)
-            script.registered_client = None  # prevent circular reference
+class ScriptManager(object):
+    def __init__(self, path):
+        self.__source = resource_string('sentry', posixpath.join('scripts', path))
+        self.__cache = WeakKeyDictionary()
+
+    def __call__(self, client, keys, args):
+        return self.for_client(client)(keys, args, client)
 
-        return script(keys, args, client)
+    def for_cluster(self, cluster):
+        return self.for_client(cluster.get_local_client(0))
```

Comment:
This method feels pretty specific. Why is it selecting the first node in the cluster? I think I'd be more in favor of this just axing this and doing the `get_local_cluster` closer to the script execution, since it's more clear what shard the script will be run on then.